pkg/telefork/client.go

```diff
@@ -0,0 +1,110 @@
+package telefork
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/getoutreach/gobox/pkg/log"
+)
+
+type Event map[string]interface{}
+
+type Client interface {
+	SendEvent(event Event)
+
+	AddField(key string, val interface{})
+	AddInfo(args ...log.Marshaler)
+
+	Close()
+}
+
+func NewClient(appName, apiKey string) Client {
+	endpoint := "https://telefork.outreach.io/"
+	if os.Getenv("OUTREACH_TELEFORK_ENDPOINT") != "" {
+		endpoint = os.Getenv("OUTREACH_TELEFORK_ENDPOINT")
+	}
+	return &client{
+		appName:  appName,
+		endpoint: endpoint,
+		apiKey:   apiKey,
+	}
+}
+
+type client struct {
+	appName  string
+	apiKey   string
+	endpoint string
+	events   []Event
+
+	commonProps map[string]interface{}
+}
+
+func (c *client) SendEvent(event Event) {
```

Comment:
Reading the code it confuses me that we have a method "SendEvent" that simply appends and "Close" that actually sends.

Reply:
Can we make it "AppendEvent" and "Flush" instead?

Reply:
This will kinda change over time. The Flush on Close is the simplest implementation. Eventually I'd like to add some reliability.

Reply:
And flush periodically not just on close.

Reply:
Understood, just reading the code on first blush was confusing though KISS makes sense

Reply:
I'll need a way to do telemetry across bin runs (one exec starts span, other exec terminates span). And there reliability will matter way more.